sway-fmt-v2/src/items/item_trait.rs

```diff
@@ -1,18 +1,116 @@
-use crate::{
-    fmt::{Format, FormattedCode, Formatter},
-    FormatterError,
-};
+use crate::{config::items::ItemBraceStyle, fmt::*, utils::bracket::CurlyBrace};
 use std::fmt::Write;
-use sway_parse::ItemTrait;
+use sway_parse::{token::Delimiter, ItemTrait, Traits};
 use sway_types::Spanned;
 
 impl Format for ItemTrait {
     fn format(
         &self,
         formatted_code: &mut FormattedCode,
-        _formatter: &mut Formatter,
+        formatter: &mut Formatter,
     ) -> Result<(), FormatterError> {
-        write!(formatted_code, "{}", self.span().as_str())?;
+        // `pub `
+        if let Some(pub_token) = &self.visibility {
+            write!(formatted_code, "{} ", pub_token.span().as_str())?;
+        }
+        // `trait name`
+        write!(
+            formatted_code,
+            "{} {} ",
+            self.trait_token.span().as_str(),
+            self.name.span().as_str()
+        )?;
+        // `: super_trait + super_trait`
+        if let Some(super_traits) = &self.super_traits {
+            formatted_code.pop(); // pop the ending space if there is a `super_trait`
```

Comment:
Rather than `pop`ping from `formatted_code` in cases like this, we might want to consider omitting the extra space from the previous formatting and re-adding it later on when necessary or writing it in an `else` block. This might make future refactoring easier if we decide to change `formatted_code` from `&mut String` to some other type of buffer or `T: fmt::Write` that does not provide a `pop` method.

Reply:
This one was an easy fix, see my comment below for more details on current status